src/Sumtypes.hs

```diff
@@ -172,8 +172,8 @@ concreteStr typeEnv env insidePath concreteStructTy@(StructTy (ConcreteNameTy ty
         )
         ( \(FuncTy [RefTy (StructTy _ _) _] StringTy _) ->
             concatMap
-              (depsOfPolymorphicFunction typeEnv env [] "prn" . typesStrFunctionType typeEnv)
-              (filter (\t -> (not . isExternalType typeEnv) t && (not . isFullyGenericType) t) (concatMap caseTys cases))
+              (depsOfPolymorphicFunction typeEnv env [] "prn" . typesStrFunctionType typeEnv env)
+              (filter (not . isFullyGenericType) (concatMap caseTys cases))
```

Comment:
I think @jacereda added a function in Utils.hs for `filter (not f)`--called `remove`

Reply:
Fixed!